### DISCUSSION TITLE
Проблема в извлечении данных из поля с типом данные

### DISCUSSION THREAD

**Discussion by RRTkornilov** [totumonline/totum-ru-issues-and-discussions#451] [Как это сделать?] [answered]
В поле c field_name и типом данные, храниться Json. В годе пытаюсь обратиться к полю и вытащить оттуда значение по ключу:
#field_name[key_name] и #field_name[key_name][sub_key_name]
Получаю ошибку что поле [[field_name]]  не найдено.
Пробовал всё, и ключи передавать в кавычках и тд.
подскажите в чем проблема и как ее решить. Заранее спасибо

**Comment by erpic-ru** [comment, marked as answer]
Пришлите пример, что именно и каким образом лежит в поле, иначе трудно сказать.
Либо там у вас массив и нужно к нему как списку обращаться через [[key]][0...], либо jsonextract какой-нибудь сделать, чтобы привести данные в структуру Тотума

**Comment by RRTkornilov** [comment]
![data](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/12359026/d97203c2-97db-453d-b52b-fca310055084)
так выглядят данные приходящие по remotes.

**Reply by erpic-ru** [reply]
обратите внимание на регистр key

**Reply by RRTkornilov** [reply]
Обращал, знаю тк немного в программировании понимаю. Но проблема остается прежней

**Reply by RRTkornilov** [reply]
#data[Name], #data["Name"] - дают ошибку: Поле [[data]] не найдено.

**Comment by poluninatm** [comment]
Перепроверьте, как вы обращаетесь к полю. Возможно, ошибка при обращении в
name поля или вы пытаетесь достать его по решетке из другой таблицы - в
случае ошибки обращения к внутренней структуре поля решетка вернула бы пустоту, а не ошибку отсутствия
поля.

**Reply by RRTkornilov** [reply]
Данные достаю из кода таблицы, где и есть это поле.

**Reply by RRTkornilov** [reply]
единственное чего не пробовал - это пересоздать таблицу)))

**Reply by poluninatm** [reply]
Таблица простая или расчетная в цикле? 
Поле в строчной части или в хэдере/футере? 
Убедитесь, что name поля точно field_name, а не h_field_name.
Вы обращаетесь из кода расчета значения другого поля или из экшена/форматирования/селекта?

**Reply by RRTkornilov** [reply]
таблица простая, поле в строчной части, имя подсвечивает редактор кода. Имя поля, специально даже создавал уникальным для теста. Обращаюсь из Экшена. 
логика такая. Приходит api запрос с данными. Я их пихаю в простую таблицу, на всякий случай. Потом экшеном на добавление строки хочу достать данные заказа и распихать по таблицам в циклах и тд.

**Reply by erpic-ru** [reply]
"Распихать по таблицам в циклах" задача не 2+2. Скорее всего, где-то там затык. Пришлите код

**Reply by RRTkornilov** [reply]
=: while(action: $addOrder;)
//СОЗДАНИЕ ЗАКАЗА В СИСТЕМЕ
addOrder: tryCatch(try: $createOrder; catch: $createException ; error: "exception") 
	createOrder: insert(table: 'orders'; field: 'order_id'= #data[Name]; inserts: "order"; log: true)

**Reply by poluninatm** [reply]
<img width="1288" alt="Снимок экрана 2023-10-05 в 18 10 31" src="https://github.com/totumonline/totum-ru-issues-and-discussions/assets/2241864/a5bfaf51-2ccd-4aa9-812c-0f00d1cc6f2c">

У меня такой код работает отлично: кнопка создает строку, экшн на добавление в поле data распихивает данные. Более дистанционно не могу предположить, что у вас могло сломаться, но со структурой поля данные это точно не связано

**Reply by poluninatm** [reply]
Попробуйте код экшена выключить при добавлении и подвесить на кнопку в той же строке: посмотрите, как добавляется информация, посмотрите логи по нажатию на кнопке.

**Comment by RRTkornilov** [comment]
Методом проверки всех возможных вариантов, нашел закономерность. Из кода экшена таблицы получить данный поля с типом данные невозможно. а вот в коде поля без проблем

**Reply by poluninatm** [reply]
Вы пробовали повторить этот пример?

https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/451#discussioncomment-7199804

Тип поля данные очень ходовой и часто используется в экшенах таблиц.

**Reply by poluninatm** [reply]
Вы пытаетесь получить данные поля в строке из кода экшена таблицы? То есть из `default_action`? Тогда нельзя получить данные из строки по решетке - не указав ид строки получить данные не из самой строки невозможно. Используйте селект или конструкцию с `@`

**Reply by erpic-ru** [reply]
Да уж, нужно было сразу спросить откуда код. Добавьте код действия в само поле с данными и все будет работать
